docs/explanation/remote-build.rst

```diff
@@ -84,37 +85,43 @@ Current
 ``--platform`` and ``--build-for``
 **********************************
 
-If  ``--platform`` or ``--build-for`` are provided, Snapcraft will:
-
-#. parse the project metadata for ``platforms`` or ``architectures`` keywords
-#. create a build plan
-#. filter the build plan with whichever of ``--build-for`` or ``--platform``
-   was specified
+``--platform`` or ``--build-for`` can only be provided when the ``platforms``
+and ``architectures`` keywords are not defined in the project metadata
+(`[12]`_).
 
 ``--build-for`` and ``--platform`` are mutually exclusive keywords.
 
-If a ``--build-for`` or ``--platform`` is provided that does not exist in the
-project metadata, Snapcraft will fail (`[1]`_).
+``core22`` snaps can only use ``--build-for``. ``core24`` and newer snaps
+can use ``--platform`` or ``--build-for``.
 
-The ``--platform`` keyword applies to ``core22`` projects because Snapcraft
-will convert ``architectures`` entries to a ``platforms`` object by using the
-``build-for`` entry of the ``architectures`` entry as the platform name.
+``--platform`` and ``--build-for`` must be a single debian architecture.
+A list of comma-separated architectures is not supported (`[11]`_).
 
 Project platforms and architectures
 ***********************************
 
 The ``snapcraft.yaml`` file is always parsed by the new remote builder.
 
+If the project metadata contains a ``platforms`` or ``architectures`` entry,
+Snapcraft will request a build for each unique ``build-for`` architecture.
+
+.. note::
+
+   Launchpad does not support cross-compiling because it does not
+   distinguish between ``build-on`` and ``build-for`` architectures (`[13]`_).
+
+.. note::
+
+    Launchpad does not support building multiple snaps on the same
+    ``build-on`` architecture (`[14]`_).
+
 If the project metadata does not contain a ``platforms`` or ``architectures``
 entry and no ``--build-for`` or ``--platform`` are passed, Snapcraft will
-request a build for the host's architecture.
+request a build on and for the host's architecture.
```

Comment:
```suggestion
request a build on, and for, the host's architecture.
```

It's a suggestion because I had to parse the sentence a couple of times to get the intended meaning